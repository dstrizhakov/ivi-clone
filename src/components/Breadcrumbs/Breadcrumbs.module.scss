@import '@/styles/variables.scss';
@import '@/styles/mixins.scss';

.container {
  margin: rem(20) 0;
}

.breadcrumbs {
  display: flex;

  &__item {
    font-size: rem(14);
    font-weight: 500;
    position: relative;
    margin-right: rem(16);
    & span {
      opacity: 0.5;
    }
    &::before {
      content: '/';
      position: absolute;
      left: rem(-11);
      color: gray;
    }
    &:first-child::before {
      content: none;
    }
    & a {
      position: relative;
      text-decoration: none;

      &::after {
        content: '';
        position: absolute;
        bottom: rem(-2);
        left: 0;
        width: 100%;
        height: rem(2);
        background-color: rgba(255, 255, 255, 0.3);

        transform-origin: left;
        transition: $time-main;
      }

      &:hover::after {
        bottom: rem(0);
        height: rem(16);
        transition: all 0.05s ease-in;
      }
    }
  }
}

.breadcrumbs__movie__item {
  float: left;
  color: rgba(white,.56);
  font-size: 15px;
  line-height: 20px;
  font-weight: 500;
  font-style: normal;
  a {
    transition-duration: .4s;
    color: rgba(white,.72);
  }
  &:hover a {
    color: white;
  }
  &.not_first:before {
    color: rgba(255,255,255,.56);
    display: inline-block;
    content: '';
    width: 2px;
    height: 2px;
    background-color: rgba(255,255,255,.56);
    margin-left: 7px;
    margin-right: 7px;
    margin-bottom: 3px;
  }
}
