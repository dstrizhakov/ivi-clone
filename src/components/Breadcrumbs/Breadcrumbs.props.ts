import { DetailedHTMLProps, HTMLAttributes } from 'react';

export type Breadcrumb = {
  name: string;
  path: string;
};

export interface BreadcrumbsProps
  extends DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
  breadcrumbs: Breadcrumb[];
}
