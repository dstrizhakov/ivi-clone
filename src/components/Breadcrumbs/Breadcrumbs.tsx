import { FC } from 'react';
import Link from 'next/link';
import styles from './Breadcrumbs.module.scss';
import { BreadcrumbsProps } from './Breadcrumbs.props';

const Breadcrumbs: FC<BreadcrumbsProps> = ({ breadcrumbs = [] }): JSX.Element => {
  return (
    <>
      {breadcrumbs.length && (
        <div className={styles.container}>
          <ul className={styles.breadcrumbs}>
            {breadcrumbs.map((breadcrumb, index) => (
              <li key={index} className={styles.breadcrumbs__item}>
                {index === breadcrumbs.length - 1 ? (
                  <span>{breadcrumb.name}</span>
                ) : (
                  <Link href={breadcrumb.path}>{breadcrumb.name}</Link>
                )}
              </li>
            ))}
          </ul>
        </div>
      )}
    </>
  );
};

export default Breadcrumbs;
