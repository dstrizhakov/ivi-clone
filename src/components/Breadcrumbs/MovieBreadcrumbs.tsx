import { FC } from 'react';
import styles from '@/components/Breadcrumbs/Breadcrumbs.module.scss';
import Link from 'next/link';
import { BreadcrumbsProps } from '@/components/Breadcrumbs/Breadcrumbs.props';

const MovieBreadcrumbs: FC<BreadcrumbsProps> = ({ breadcrumbs = [] }): JSX.Element => {
  return (
    <div className={styles.container}>
      <ul className={styles.breadcrumbs}>
        {breadcrumbs.map((breadcrumb, index) => (
          <li
            key={index}
            className={`${styles.breadcrumbs__movie__item} ${index && styles.not_first}`}
          >
            <Link href={breadcrumb.path}>{breadcrumb.name}</Link>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default MovieBreadcrumbs;
