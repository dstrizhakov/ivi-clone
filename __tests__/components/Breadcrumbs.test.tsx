import { render } from '@testing-library/react';
import React from 'react';
import Breadcrumbs from '../../src/components/Breadcrumbs/Breadcrumbs';
import MovieBreadcrumbs from '../../src/components/Breadcrumbs/MovieBreadcrumbs';

describe('breadcrumbs', () => {
  it('should be correct with 3 breadcrumbs', () => {
    const breadcrumbsMockData = [
      { name: 'breadcrumb1', path: '/' },
      { name: 'breadcrumb2', path: '/movies' },
      { name: 'breadcrumb3', path: '/movies' },
    ];
    const bc = render(<Breadcrumbs breadcrumbs={breadcrumbsMockData} />);
    expect(bc).toMatchSnapshot();
  });
  it('should be correct with 2 breadcrumbs', () => {
    const breadcrumbsMockData = [
      { name: 'breadcrumb1', path: '/' },
      { name: 'breadcrumb2', path: '/movies' },
    ];
    const bc = render(<Breadcrumbs breadcrumbs={breadcrumbsMockData} />);
    expect(bc).toMatchSnapshot();
  });
  it('should be correct with 1 breadcrumbs', () => {
    const breadcrumbsMockData = [{ name: 'breadcrumb1', path: '/' }];
    const bc = render(<Breadcrumbs breadcrumbs={breadcrumbsMockData} />);
    expect(bc).toMatchSnapshot();
  });
  it('should be correct with no breadcrumbs', () => {
    const breadcrumbsMockData = [];
    const bc = render(<Breadcrumbs breadcrumbs={breadcrumbsMockData} />);
    expect(bc).toMatchSnapshot();
  });
});

describe("movie's breadcrumbs", () => {
  it('should be correct with 3', () => {
    const breadcrumbsMockData = [
      { name: 'breadcrumb1', path: '/' },
      { name: 'breadcrumb2', path: '/movies' },
      { name: 'breadcrumb3', path: '/movies' },
    ];
    const bc = render(<MovieBreadcrumbs breadcrumbs={breadcrumbsMockData} />);
    expect(bc).toMatchSnapshot();
  });
  it('should be correct with 2', () => {
    const breadcrumbsMockData = [
      { name: 'breadcrumb1', path: '/' },
      { name: 'breadcrumb2', path: '/movies' },
    ];
    const bc = render(<MovieBreadcrumbs breadcrumbs={breadcrumbsMockData} />);
    expect(bc).toMatchSnapshot();
  });
  it('should be correct with 1', () => {
    const breadcrumbsMockData = [{ name: 'breadcrumb1', path: '/' }];
    const bc = render(<MovieBreadcrumbs breadcrumbs={breadcrumbsMockData} />);
    expect(bc).toMatchSnapshot();
  });
  it('should be correct with undefined', () => {
    const breadcrumbsMockData = undefined;
    const bc = render(<MovieBreadcrumbs breadcrumbs={breadcrumbsMockData} />);
    expect(bc).toMatchSnapshot();
  });
});
