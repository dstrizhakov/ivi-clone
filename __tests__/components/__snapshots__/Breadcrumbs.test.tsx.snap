// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumbs should be correct with 1 breadcrumbs 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <ul
          class="breadcrumbs"
        >
          <li
            class="breadcrumbs__item"
          >
            <span>
              breadcrumb1
            </span>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <ul
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__item"
        >
          <span>
            breadcrumb1
          </span>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`breadcrumbs should be correct with 2 breadcrumbs 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <ul
          class="breadcrumbs"
        >
          <li
            class="breadcrumbs__item"
          >
            <a
              href="/"
            >
              breadcrumb1
            </a>
          </li>
          <li
            class="breadcrumbs__item"
          >
            <span>
              breadcrumb2
            </span>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <ul
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__item"
        >
          <a
            href="/"
          >
            breadcrumb1
          </a>
        </li>
        <li
          class="breadcrumbs__item"
        >
          <span>
            breadcrumb2
          </span>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`breadcrumbs should be correct with 3 breadcrumbs 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <ul
          class="breadcrumbs"
        >
          <li
            class="breadcrumbs__item"
          >
            <a
              href="/"
            >
              breadcrumb1
            </a>
          </li>
          <li
            class="breadcrumbs__item"
          >
            <a
              href="/movies"
            >
              breadcrumb2
            </a>
          </li>
          <li
            class="breadcrumbs__item"
          >
            <span>
              breadcrumb3
            </span>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <ul
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__item"
        >
          <a
            href="/"
          >
            breadcrumb1
          </a>
        </li>
        <li
          class="breadcrumbs__item"
        >
          <a
            href="/movies"
          >
            breadcrumb2
          </a>
        </li>
        <li
          class="breadcrumbs__item"
        >
          <span>
            breadcrumb3
          </span>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`breadcrumbs should be correct with no breadcrumbs 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      0
    </div>
  </body>,
  "container": <div>
    0
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`movie's breadcrumbs should be correct with 1 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <ul
          class="breadcrumbs"
        >
          <li
            class="breadcrumbs__movie__item 0"
          >
            <a
              href="/"
            >
              breadcrumb1
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <ul
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__movie__item 0"
        >
          <a
            href="/"
          >
            breadcrumb1
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`movie's breadcrumbs should be correct with 2 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <ul
          class="breadcrumbs"
        >
          <li
            class="breadcrumbs__movie__item 0"
          >
            <a
              href="/"
            >
              breadcrumb1
            </a>
          </li>
          <li
            class="breadcrumbs__movie__item not_first"
          >
            <a
              href="/movies"
            >
              breadcrumb2
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <ul
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__movie__item 0"
        >
          <a
            href="/"
          >
            breadcrumb1
          </a>
        </li>
        <li
          class="breadcrumbs__movie__item not_first"
        >
          <a
            href="/movies"
          >
            breadcrumb2
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`movie's breadcrumbs should be correct with 3 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <ul
          class="breadcrumbs"
        >
          <li
            class="breadcrumbs__movie__item 0"
          >
            <a
              href="/"
            >
              breadcrumb1
            </a>
          </li>
          <li
            class="breadcrumbs__movie__item not_first"
          >
            <a
              href="/movies"
            >
              breadcrumb2
            </a>
          </li>
          <li
            class="breadcrumbs__movie__item not_first"
          >
            <a
              href="/movies"
            >
              breadcrumb3
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <ul
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__movie__item 0"
        >
          <a
            href="/"
          >
            breadcrumb1
          </a>
        </li>
        <li
          class="breadcrumbs__movie__item not_first"
        >
          <a
            href="/movies"
          >
            breadcrumb2
          </a>
        </li>
        <li
          class="breadcrumbs__movie__item not_first"
        >
          <a
            href="/movies"
          >
            breadcrumb3
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`movie's breadcrumbs should be correct with undefined 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <ul
          class="breadcrumbs"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <ul
        class="breadcrumbs"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
